pkg/oc/cli/admin/migrate/migrator.go

```diff
@@ -135,6 +138,38 @@ type ResourceOptions struct {
 	genericclioptions.IOStreams
 }
 
+func NewResourceOptions(streams genericclioptions.IOStreams) *ResourceOptions {
+	return &ResourceOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("migrated").WithTypeSetter(scheme.Scheme),
```

Comment:
What is this?

Reply:
New printing stack. We have been pushing the use of a single code-path (this one) to print anything in the cli.  The `genericclioptions` package was recently moved to its own repo upstream (`k8s.io/cli-runtime`) which means that `oc` will become a primary consumer of that package.

The main gist behind the new printing stack is that it requires all consumers to pass it external objects only. It also enforces additional metadata for each object that it receives, so that anything that is printed to the user is displayed accurately, and in a format consumable by other `oc` commands. We start out with a struct that the command can use to bind all printing-related flags at once. Then from that struct, use flag values provided by the user to generate the appropriate printer.

The type-setter is temporary, but for now it allows us to attempt to retrieve the gvk for runtime.Objects that may not have it set.